src/Microsoft.Performance.SDK.Runtime/Extensibility/DataExtensions/Tables/TableExtensionReference.cs

```diff
@@ -146,15 +149,18 @@ public TableDescriptor TableDescriptor
         /// </exception>
         internal static bool TryCreateReference(
             Type candidateType,
+            ILogger logger,
             out ITableExtensionReference reference)
         {
-            Guard.NotNull(candidateType, nameof(candidateType));
+            Debug.Assert(candidateType != null, $"{nameof(candidateType)} cannot be null.");
```

Comment:
Why change these to debug asserts?

Reply:
It's on an internal method. We only throw on publicly facing APIs. We Debug.Assert if it's internal to our code and only we should be calling it.